interoperability_report.py

```diff
@@ -822,120 +528,168 @@ def parser():
-        # apart from other custom attributes (in this case param_pub,
-        # param_sub, param_pub1 and param_pub2).
-        case = TestCase(f'{k}')
-        now_test_case = datetime.now()
-        if k == 'Test_Ownership_3' or k == 'Test_Ownership_4':
-            run_test_pub_pub_sub(name_executable_pub=options['publisher'],
-                                 name_executable_sub=options['subscriber'],
-                                 test_case=case,
-                                 param_pub1=v[0],
-                                 param_pub2=v[1],
-                                 param_sub=v[2],
-                                 expected_code_pub1=v[3],
-                                 expected_code_pub2=v[4],
-                                 expected_code_sub=v[5],
-                                 verbosity=options['verbosity'],
-                                 timeout=timeout
-            )
 
-        else:
-            run_test(name_executable_pub=options['publisher'],
-                     name_executable_sub=options['subscriber'],
-                     test_case=case,
-                     param_pub=v[0],
-                     param_sub=v[1],
-                     expected_code_pub=v[2],
-                     expected_code_sub=v[3],
-                     verbosity=options['verbosity'],
-                     timeout=timeout
-            )
-        case.time = (datetime.now() - now_test_case).total_seconds()
-        suite.add_testcase(case)
+    for element in options['test_suite']:
+        t_suite_module = importlib.import_module(element,'./')
+        for name, t_suite in inspect.getmembers(t_suite_module):
+            # we take all the dictionaries defined in the test_suite file except __builtins__.
+            if type(t_suite) is dict and name != '__builtins__':
+                # check that the test_cases selected are in the test_suite
+                check_test_case_in_test_suite(t_suite, name, options['test_cases'])
+                check_test_case_in_test_suite(t_suite, name, options['test_cases_disabled'])
+
+                for k, v in t_suite.items():
+                    # TestCase is a class from junitparser whose attributes
+                    # are: name and result (OK, Failure, Error or Skipped).
+                    parameters = v[0]
+                    expected_codes = v[1]
+                    if len(v) == 3:
+                        check_function = v[2]
+                    elif len(v) == 2:
+                        check_function = no_check
+                    else:
+                        print('Error in the definition of the Test Suite. Number of arguments incorrect.')
+                        break
+                    assert(len(parameters) == len(expected_codes))
+                    if (options['test_cases'] == None or k in options['test_cases']) \
+                        and (options['test_cases_disabled'] == None or k not in options['test_cases_disabled']):
```

Comment:
```suggestion
                    if len(v) == 3:
                        check_function = v[2]
                    elif len(v) == 2:
                        check_function = no_check
                    else:
                        print('Error in the definition of the Test Suite. Number "...
                            "of arguments incorrect.')
                        break

                    assert(len(parameters) == len(expected_codes))

                    # Check that the test is not disabled or it is part of the enabled test list
                    if (options['test_cases'] == None or k in options['test_cases'])
                            and (options['test_cases_disabled'] == None
                                or k not in options['test_cases_disabled']):
```

syntax may not be correct